packages/webapp/src/containers/Insights/Prices/index.js

```diff
@@ -46,44 +45,60 @@ class Prices extends Component {
   }
 
   componentDidUpdate(prevProps) {
-    if (prevProps.pricesDistance !== this.props.pricesDistance && this.props.farm.grid_points) {
-      this.props.dispatch(getPricesWithDistanceData(this.props.farm.grid_points, this.props.pricesDistance))
+    if (prevProps.pricesDistance !== this.props.pricesDistance) {
+      if(this.props.farm.grid_points){
+        this.props.dispatch(getPricesWithDistanceData(this.props.farm.grid_points, this.props.pricesDistance));
+        //TODO: need rewrite
+      }
+    }
+    if(this.props.farm?.units?.measurement === 'imperial' && Number.isInteger(this.state.distance)){
+      console.log('imperial', this.props.farm?.units?.measurement);
+      this.handleChange(5*1.609);
+    }else if(this.props.farm?.units?.measurement === 'metric' && !Number.isInteger(this.state.distance)){
+      console.log('metric', this.props.farm?.units?.measurement);
+      this.handleChange(5);
     }
   }
 
   render() {
+    const isImperial = this.props.farm && this.props.farm.units?.measurement === 'imperial';
+    const distances = isImperial ? [5, 10, 20, 35] : [5, 10, 25, 50]
     const { currencySymbol } = this.state;
     const { pricesData } = this.props;
     const { data: cropsWithPriceInfo } = pricesData;
     return (
       <div className={insightStyles.insightContainer}>
         <PageTitle title="Prices" backUrl="/Insights/"
-                   rightIcon={true} rightIconBody={this.infoBoxBody} rightIconTitle={"Prices"}/>
+                   rightIcon={true} rightIconBody={this.infoBoxBody} rightIconTitle={'Prices'}/>
         {this.state.distance &&
         <div>
-          <div style={{float: 'left'}}><b>Sales from {this.state.distance} km away</b></div>
-          <div style={{float: 'right'}}><PriceDistanceComponent handleOpenCollapse={this.handleOpenCollapse}/></div>
+          <div style={{ float: 'left' }}><b>Sales from {isImperial? this.state.distance/1.609 : this.state.distance} {isImperial? 'mi':'km'} away</b></div>
```

Comment:
Can we move 1.609 to a constant in the start of the file, it will be a little more readable this way